tests/fast-integration/skr-test/btp-manager-secret/index.js

```diff
@@ -0,0 +1,115 @@
+const {expect} = require('chai');
+const {
+  getSecret,
+  getSecretData,
+  k8sDelete,
+  waitForSecret,
+  k8sApply,
+  waitForK8sObject,
+} = require('../../utils');
+const {BTPOperatorCreds} = require('../../smctl/helpers');
+
+const secretName = 'sap-btp-manager';
+const ns = 'kyma-system';
+const expectedBtpOperatorCreds = BTPOperatorCreds.dummy();
+
+const reconciliationTimeout = 1000 * 70;
+let secretFromProvisioning;
+let modifiedSecret;
+
+function btpManagerSecretTest() {
+  describe('BTP Manager Secret Test', function() {
+    // Check if BTP Manager Secret with BTP Operator credentials is created properly
+    it('should check if Secret exists and contains the expected data keys', async function() {
+      secretFromProvisioning = await getSecret(secretName, ns);
+      checkSecretDataKeys(secretFromProvisioning);
+      modifiedSecret = JSON.parse(JSON.stringify(secretFromProvisioning));
+    });
+    // Check if the Secret contains expected values
+    it('should check if Secret data values match expected values', async function() {
+      const actualSecretData = await getSecretData(secretName, ns);
+      checkSecretDataValues(actualSecretData);
+    });
+    // Check if the Secret is properly reconciled after deletion
+    it('should check if Secret is reconciled after deletion', async function() {
+      console.log(`Deleting the "sap-btp-manager" Secret`);
+      await k8sDelete([secretFromProvisioning], ns);
+      console.log(`Waiting for the reconciliation for ${reconciliationTimeout} ms`);
+      await waitForSecret(secretName, ns, reconciliationTimeout);
+      console.log(`Secret has been re-created. Checking Secret's data`);
+      const actualSecret = await getSecret(secretName, ns);
+      checkSecretDataKeys(actualSecret);
+      const actualSecretData = await getSecretData(secretName, ns);
+      checkSecretDataValues(actualSecretData);
+      console.log(`Secret has been properly reconciled`);
+    });
+    // Check if the Secret is properly reconciled after being edited
+    it('should check if Secret is reconciled after being edited', async function() {
+      console.log(`Changing data in the "sap-btp-manager" Secret`);
+      prepareSecretForApply(modifiedSecret);
+      changeSecretData(modifiedSecret);
+      console.log(`Applying edited Secret`);
+      await k8sApply([modifiedSecret], ns);
+      console.log(`Waiting ${reconciliationTimeout} ms until edited Secret is created`);
+      await waitForSecret(secretName, ns, reconciliationTimeout);
+      let actualSecret = await getSecret(secretName, ns);
+      console.log(`Waiting for the reconciliation for ${reconciliationTimeout} ms`);
+      await waitForK8sObject(
```

Comment:
Why don't we extract this part as a separate function, like we do with `waitForSecret`. 
Then we would have the same level of abstraction (more or less) in this assertion.

Reply:
I used a closure instead of a defined function to avoid defining what we can treat as a Secret reconciliation. In this case, we know that KEB is the only entity to modify the Secret, but resourceVersion can change due to user actions like patch or update and it won't be a reconciliation. If I would define a func like `waitForSecretReconcile` other developers could assume that the func always waits for a reconciliation when it doesn't.